enarx-keep-sev-lib/src/proxy.rs

```diff
@@ -5,120 +5,81 @@
-    /// calling convention.
-    pub rax: Register<u64>,
-    /// 1st parameter register.
-    pub rdi: Register<u64>,
-    /// 2nd parameter register.
-    pub rsi: Register<u64>,
-    /// 3rd parameter register.
-    pub rdx: Register<u64>,
-    /// 4th parameter register.
-    pub r10: Register<u64>,
-    /// 5th parameter register.
-    pub r8: Register<u64>,
-    /// 6th parameter register.
-    pub r9: Register<u64>,
-}
-
-impl Message {
-    /// Construct a new `Message`.
-    pub fn new(rax: Register<u64>) -> Self {
-        Self {
-            rax,
-            rdi: Register::from_raw(0),
-            rsi: Register::from_raw(0),
-            rdx: Register::from_raw(0),
-            r10: Register::from_raw(0),
-            r8: Register::from_raw(0),
-            r9: Register::from_raw(0),
-        }
-    }
-
-    /// Assign a value to the first parameter register (rdi).
-    pub fn first(mut self, val: Register<u64>) -> Self {
-        self.rdi = val;
-        self
-    }
-
-    /// Access the first parameter register (rdi).
-    pub fn get_first(&self) -> Register<u64> {
-        self.rdi
-    }
+    /// See, for example, libc::SYS_exit.
+    pub num: Register<usize>,
 
-    /// Assign a value to the second parameter register (rsi).
-    pub fn second(mut self, val: Register<u64>) -> Self {
-        self.rsi = val;
-        self
-    }
-
-    /// Access the second parameter register (rsi).
-    pub fn get_second(&self) -> Register<u64> {
-        self.rsi
-    }
-
-    /// Assign a value to the third parameter register (rdx).
-    pub fn third(mut self, val: Register<u64>) -> Self {
-        self.rdx = val;
-        self
-    }
-
-    /// Access the third parameter register (rdx).
-    pub fn get_third(&self) -> Register<u64> {
-        self.rdx
-    }
-
-    /// Assign a value to the fourth parameter register (r10).
-    pub fn fourth(mut self, val: Register<u64>) -> Self {
-        self.r10 = val;
-        self
-    }
+    /// The syscall argument registers
+    ///
+    /// At most 6 syscall arguments can be provided.
+    pub arg: [Register<usize>; 6],
+}
 
-    /// Access the fourth parameter register (r10)
-    pub fn get_fourth(&self) -> Register<u64> {
-        self.r10
-    }
+/// A reply
+///
+/// The `Reply` struct is the most minimal representation of the register context
+/// needed for service replies from the hypervisor to the microkernel. An example
+/// of such a reply would be the return value from a proxied system call.
+///
+/// Although most architectures collapse this to a single register value
+/// with error numbers above `usize::max_value() - 4096`, `ppc64` uses
+/// the `cr0.SO` flag to indicate error instead. Unfortunately, we also
+/// can't use the built-in `Result` type for this, since its memory layout
+/// is undefined. Therefore, we use this layout with conversions for `Result`.
+#[repr(C)]
+#[derive(Copy, Clone, Default)]
+pub struct Reply {
+    val: Register<usize>,
+    err: bool,
+}
```

Comment:
If we're establishing a proxy interface for more than just SEV then we should probably revert e527737a0eef01fb8d6d5b8cb37edce7700575e0.

Reply:
We might even use this in SGX. So perhaps we should name it something completely different than `vmsyscall`.

Reply:
How about `sally-port`? [Sally Port](https://en.wikipedia.org/wiki/Sally_port) Or is that too much :sweat_smile:

Reply:
@connorkuehl `sallyport` it is! At some point we're going to jump the shark. Today is not that day. :)